### PR TITLE
[#100] Add rest server basic sample code.

### LINKED ISSUE #100: Add rest server basic sample code to dss-example module
- create io.github.ztkmkoo.dss.example.http.tutorial package under dss-example module
- complete sample code from https://github.com/ztkmkoo/dss/wiki/Project-Tuturial(KR)

### PR DESCRIPTION
**Pull request by RulLu16**
Resolves #100 

Add rest server basic sample code to dss-example module.

- I make io.github.ztkmkoo.dss.example.http.tutorial package in example module and add tutorial code in it.

### CODE REVIEW

**Comment by ztkmkoo** [comment]
@RulLu16 
Hi, i have a question that the test coverage will be reduced when i add new sample code without test?

If true, we have exclude the sample module when coveralls aggregates the test coverage.

**Comment by RulLu16** [comment]
> @RulLu16
> Hi, i have a question that the test coverage will be reduced when i add new sample code without test?
> 
> If true, we have exclude the sample module when coveralls aggregates the test coverage.

@ztkmkoo 
I don't fully understand what is problem.. So, you think we have to make this sample code to test code or add test code?
I think we don't have to do that. Some other opensource project, like Netty, just use example code without using test. 

But if you think it can cause problems later, I'm willing to fix it.

**Comment by RulLu16** [comment]
@Doyuni @ParkSale 
Thank you for review! :)

**Comment by ztkmkoo** [comment]
> 
> 
> > @RulLu16
> > Hi, i have a question that the test coverage will be reduced when i add new sample code without test?
> > If true, we have exclude the sample module when coveralls aggregates the test coverage.
> 
> @ztkmkoo
> I don't fully understand what is problem.. So, you think we have to make this sample code to test code or add test code?
> I think we don't have to do that. Some other opensource project, like Netty, just use example code without using test.
> 
> But if it is likely to cause problems later, I'm willing to fix it.

@RulLu16 
Hi, 
It is inefficient to create tests of these sample codes. It must be duplicate job, i think.
Simplest way is exclude sample code test coverage check from coveralls report. (I have no idea if we can exclude it)

@RulLu16 @Alencion @ParkSale @Doyuni 
Is that the best way?

**Comment by Alencion** [comment]
> > > @RulLu16
> > > Hi, i have a question that the test coverage will be reduced when i add new sample code without test?
> > > If true, we have exclude the sample module when coveralls aggregates the test coverage.
> > 
> > 
> > @ztkmkoo
> > I don't fully understand what is problem.. So, you think we have to make this sample code to test code or add test code?
> > I think we don't have to do that. Some other opensource project, like Netty, just use example code without using test.
> > But if it is likely to cause problems later, I'm willing to fix it.
> 
> @RulLu16
> Hi,
> It is inefficient to create tests of these sample codes. It must be duplicate job, i think.
> Simplest way is exclude sample code test coverage check from coveralls report. (I have no idea if we can exclude it)
> 
> @RulLu16 @Alencion @ParkSale @Doyuni
> Is that the best way?

I also think it's overlapping. So it's better to find a way to exclude specific folders.

**Comment by RulLu16** [comment]
> > > @RulLu16
> > > Hi, i have a question that the test coverage will be reduced when i add new sample code without test?
> > > If true, we have exclude the sample module when coveralls aggregates the test coverage.
> > 
> > 
> > @ztkmkoo
> > I don't fully understand what is problem.. So, you think we have to make this sample code to test code or add test code?
> > I think we don't have to do that. Some other opensource project, like Netty, just use example code without using test.
> > But if it is likely to cause problems later, I'm willing to fix it.
> 
> @RulLu16
> Hi,
> It is inefficient to create tests of these sample codes. It must be duplicate job, i think.
> Simplest way is exclude sample code test coverage check from coveralls report. (I have no idea if we can exclude it)
> 
> @RulLu16 @Alencion @ParkSale @Doyuni
> Is that the best way?

I also think we don't have to make test codes. It's more efficient to exclude this one from test coverage. 

I'll try this 
https://stackoverflow.com/questions/34699240/instruct-jacoco-and-coveralls-to-ignore-some-files-from-the-report

**Comment by ztkmkoo** [comment]
> I'll try this
> https://stackoverflow.com/questions/34699240/instruct-jacoco-and-coveralls-to-ignore-some-files-from-the-report

It looks nice to exclude from jacoco report. Thanks

**Comment by ztkmkoo** [comment]
> 
> 
> 해당 예제외에 다른 입력도 고려해주시면 좋을 듯 합니다.
> 
> 가령
> 
> ```
> {
> 	"name": "hello",
> 	"test": ["a", "b", "c"]
> }
> ```
> 
> 및
> 
> ```
> {
> 	"name": "hello",
> 	"test1": {
> 		"test2": {
>                  }
> 	}
> }
> ```
> 
> 이런 입력에 대한 처리방법도 샘플로 달아주시면 좋을듯 합니다.

@RulLu16 @Alencion @ParkSale @vincentj2 @Doyuni @Amazansia 
Do we need some complex(?) request for this sample code? Can we discuss it?

File: pom.xml [hunk 1]
```diff
@@ -121,6 +121,11 @@
                 <groupId>org.eluder.coveralls</groupId>
                 <artifactId>coveralls-maven-plugin</artifactId>
                 <version>${coveralls.version}</version>
+                <configuration>
+                    <excludes>
+                        <exclude>io.github.ztkmkoo.dss.example/**/*</exclude>
+                    </excludes>
+                </configuration>
```
Comment (ztkmkoo) [inline pom.xml:128]: Do you mean that you want to exclude test from jacoco report?
Ref https://github.com/trautonen/coveralls-maven-plugin.
The plugin for submitting Java code coverage reports to Coveralls web service. == only submitt jacoco report to coveralls

Comment (RulLu16) [reply pom.xml:128]: I just think that there are no @test in example, so jacoco reports is already skipped. So I insert in coveralls part. 

And it's my bad. I apologize. Thanks to your ref, I see that inserting configuration to coveralls is not good. So I'll move this exclude configuration to jacoco now.

**Comment by ztkmkoo** [comment]
@RulLu16 
Hi, i'm so sorry..

I found if no test in the module, jacoco not aggregate the test coverage..

### the case with example module
```
[INFO] Gathered code coverage metrics for 42 source files with 1850 lines of code:
[INFO] - 573 relevant lines
[INFO] - 481 covered lines
[INFO] - 92 missed lines
```

### the case without example module
```
[INFO] Gathered code coverage metrics for 42 source files with 1850 lines of code:
[INFO] - 573 relevant lines
[INFO] - 481 covered lines
[INFO] - 92 missed lines
```

Both cases have same test coverage report.

**Pull request merged**